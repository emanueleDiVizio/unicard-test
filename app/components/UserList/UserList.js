import React from 'react';
import {View, Text, StyleSheet, Image, FlatList} from 'react-native';
import useFetch from 'react-fetch-hook';

const UserItem = ({image, name, surname, email}) => {
  return (
    <View style={styles.container}>
      <Image source={{uri: image}} style={styles.image} />
      <View style={styles.textContainer}>
        <Text style={styles.name}>
          {name} {surname}
        </Text>
        <Text>{email}</Text>
      </View>
    </View>
  );
};

const UserList = () => {
  const {isLoading, data} = useFetch('https://reqres.in/api/users');

  return isLoading ? (
    <></>
  ) : (
    <FlatList
      data={data.data}
      renderItem={({item: {avatar, first_name, last_name, email}}) => (
        <UserItem
          image={avatar}
          name={first_name}
          surname={last_name}
          email={email}
        />
      )}
      keyExtractor={({email}) => email}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    margin: 8,
  },
  image: {
    width: 100,
    height: 100,
  },
  textContainer: {
    marginLeft: 16,
    marginTop: 8,
  },
  name: {fontWeight: 'bold', fontSize: 16},
});

export default UserList;
