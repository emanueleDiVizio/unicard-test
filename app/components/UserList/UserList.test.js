import React from 'react';
import UserList from './UserList';
import renderer from 'react-test-renderer';
import useFetch from 'react-fetch-hook';

jest.mock('react-fetch-hook', () => ({
  __esModule: true,
  default: jest.fn(),
}));

const MOCK_DATA = [
  {
    id: 1,
    email: 'george.bluth@reqres.in',
    first_name: 'George',
    last_name: 'Bluth',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg',
  },
  {
    id: 2,
    email: 'janet.weaver@reqres.in',
    first_name: 'Janet',
    last_name: 'Weaver',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/josephstein/128.jpg',
  },
  {
    id: 3,
    email: 'emma.wong@reqres.in',
    first_name: 'Emma',
    last_name: 'Wong',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/olegpogodaev/128.jpg',
  },
  {
    id: 4,
    email: 'eve.holt@reqres.in',
    first_name: 'Eve',
    last_name: 'Holt',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg',
  },
  {
    id: 5,
    email: 'charles.morris@reqres.in',
    first_name: 'Charles',
    last_name: 'Morris',
    avatar:
      'https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg',
  },
  {
    id: 6,
    email: 'tracey.ramos@reqres.in',
    first_name: 'Tracey',
    last_name: 'Ramos',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg',
  },
];

describe('UserList', () => {
  it('displays nothing when loading', () => {
    useFetch.mockImplementationOnce(() => ({
      data: {data: []},
      isLoading: true,
    }));
    const element = renderer.create(<UserList />).toJSON();

    expect(element).toMatchSnapshot();
  });

  it('displays user list when data is loaded', () => {
    useFetch.mockImplementationOnce(() => ({
      data: {
        data: MOCK_DATA,
      },
      isLoading: false,
    }));
    const element = renderer.create(<UserList />).toJSON();

    expect(element).toMatchSnapshot();
  });
});
