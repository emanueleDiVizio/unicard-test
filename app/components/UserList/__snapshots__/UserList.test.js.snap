// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList displays nothing when loading 1`] = `null`;

exports[`UserList displays user list when data is loaded 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
        "email": "george.bluth@reqres.in",
        "first_name": "George",
        "id": 1,
        "last_name": "Bluth",
      },
      Object {
        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/josephstein/128.jpg",
        "email": "janet.weaver@reqres.in",
        "first_name": "Janet",
        "id": 2,
        "last_name": "Weaver",
      },
      Object {
        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/olegpogodaev/128.jpg",
        "email": "emma.wong@reqres.in",
        "first_name": "Emma",
        "id": 3,
        "last_name": "Wong",
      },
      Object {
        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
        "email": "eve.holt@reqres.in",
        "first_name": "Eve",
        "id": 4,
        "last_name": "Holt",
      },
      Object {
        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
        "email": "charles.morris@reqres.in",
        "first_name": "Charles",
        "id": 5,
        "last_name": "Morris",
      },
      Object {
        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
        "email": "tracey.ramos@reqres.in",
        "first_name": "Tracey",
        "id": 6,
        "last_name": "Ramos",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 16,
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            George
             
            Bluth
          </Text>
          <Text>
            george.bluth@reqres.in
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/josephstein/128.jpg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 16,
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Janet
             
            Weaver
          </Text>
          <Text>
            janet.weaver@reqres.in
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/olegpogodaev/128.jpg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 16,
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Emma
             
            Wong
          </Text>
          <Text>
            emma.wong@reqres.in
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 16,
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Eve
             
            Holt
          </Text>
          <Text>
            eve.holt@reqres.in
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 16,
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Charles
             
            Morris
          </Text>
          <Text>
            charles.morris@reqres.in
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 16,
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Tracey
             
            Ramos
          </Text>
          <Text>
            tracey.ramos@reqres.in
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
